### PR TITLE
Don't use a definition file for the bindings.

### PR DESCRIPTION
**Pull request by nkreeger**
This causes many issues and doesn't help with redistribution. It looks
like exporting a local definition file on top of what else is exported
is not a common supported TypeScript use case. This fix simply moves
defnitions into a normal TypeScript file.

This should fix: https://github.com/tensorflow/tfjs/issues/1092

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-node/224)

### CODE REVIEW

File: src/tfjs_binding.ts [hunk 1]
```diff
@@ -15,13 +15,13 @@
  * =============================================================================
  */
 
-declare class TensorMetadata {
+export class TensorMetadata {
```
Comment (dsmilkov) [inline src/tfjs_binding.ts]: you can keep the `declare` so you have `export declare`. This tells external tools like Closure that the object comes from an external source so it doesn't rewrite it.

Comment (dsmilkov) [reply src/tfjs_binding.ts]: You might have to `s/class/interface`

**Comment by nsthorat** [comment]
It would be good in this PR to have a simple integration test that publishes node with yalc and then links it in a directory that simply compiles an external project that depends on tfjs-node@master. This way this problem wont happen again.

**Comment by nkreeger** [comment]
@nsthorat @dsmilkov PTAL - I added a very small internal project to test this (rather than create a stub in a new repo).

The new integration test does:
- Preps and yalc's the tfjs-node package
- cd's into internal project
- Preps the environment, ensures the binding is compiled (yalc won't do this)
- Executes the sample typescript app
- Runs the tsc compiler
- Executes the compiled JS version of app

**Pull request merged**